docassemble/AssemblyLine/al_document.py

```diff
@@ -1489,38 +1537,31 @@ def _update_labels(self, auto_labeler: Callable = None) -> None:
 
     def ocr_ready(self) -> bool:
         """
-        Returns:
-            True iff OCR process has finished on all pages. OCR is non-blocking, and assembly will work
-            even if OCR is not complete. Check this status if you want to wait to deliver a document until
-            OCR is complete.
+        Combines the results of all the ALExhibit.ocr_ready calls.
+        See ALExhibi.ocr_ready for more details.
         """
         ready = True
         for exhibit in self.elements:
-            for page in exhibit.pages:
-                if hasattr(page, "ocr_status"):
-                    ready &= page.ocr_status.ready()
+            ready &= exhibit.ocr_ready()
         return ready
 
-    def _update_page_numbers(self) -> None:
+    def _update_page_numbers(
+        self, starting_number: Optional[int] = None, toc_guess_pages: int = 1
+    ) -> None:
         """
         Update the `start_page` attribute of all exhibits so it reflects current position in the list + number of pages of each document.
         """
-        current_index = 1
+        toc_pages = toc_guess_pages if self.include_table_of_contents else 0
```

Comment:
We already do this (and tested! It takes 28 attachments to overflow the ToC page, lol) [in the ALExhibitDocument](https://github.com/SuffolkLITLab/docassemble-AssemblyLine/pull/508/files#diff-db23667d014599821c0250a04d5ffbbcf327b89266e017c6f6304c30a19ff8efR1672), where we do have the table of contents document itself.  We can't do that here because we don't have the toc here.